internal/xds/rbac/rbac_engine.go

```diff
@@ -238,3 +275,42 @@ type rpcData struct {
 	// handshake.
 	certs []*x509.Certificate
 }
+
+func (e *engine) doAuditLogging(rpcData *rpcData, rule string, authorized bool) {
+	// In the RBAC world, we need to have a SPIFFE ID as the principal for this
+	// to be meaningful
+	principal := ""
+	if rpcData.peerInfo != nil && rpcData.peerInfo.AuthInfo != nil && rpcData.peerInfo.AuthInfo.AuthType() == "tls" {
+		// If AuthType = tls, then we can cast AuthInfo to TLSInfo.
+		tlsInfo := rpcData.peerInfo.AuthInfo.(credentials.TLSInfo)
+		if tlsInfo.SPIFFEID != nil {
+			principal = tlsInfo.SPIFFEID.String()
+		}
+	}
+
+	event := &audit.Event{
+		FullMethodName: rpcData.fullMethod,
+		Principal:      principal,
+		PolicyName:     e.policyName,
+		MatchedRule:    rule,
+		Authorized:     authorized,
+	}
+	for _, logger := range e.auditLoggers {
+		switch e.auditCondition {
```

Comment:
We don't have to do it in this PR. But as an optimization, it might be better to check whether we want to log first before we prepare the `Event`, this is within the RPC flow, so any effort we can save is good. If you could add a TODO it'd be good enough.

Reply:
Added